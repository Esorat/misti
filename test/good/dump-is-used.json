{
  "projectName": "dump-is-used",
  "functions": [
    {
      "name": "test1",
      "cfg": {
        "nodes": [
          {
            "id": 141,
            "stmtID": 1654,
            "srcEdges": [],
            "dstEdges": [143]
          },
          {
            "id": 142,
            "stmtID": 1658,
            "srcEdges": [143],
            "dstEdges": [145]
          },
          {
            "id": 144,
            "stmtID": 1662,
            "srcEdges": [145],
            "dstEdges": [147]
          },
          {
            "id": 146,
            "stmtID": 1666,
            "srcEdges": [147],
            "dstEdges": [149]
          },
          {
            "id": 148,
            "stmtID": 1670,
            "srcEdges": [149],
            "dstEdges": [151]
          },
          {
            "id": 150,
            "stmtID": 1674,
            "srcEdges": [151],
            "dstEdges": []
          }
        ],
        "edges": [
          {
            "id": 143,
            "src": 141,
            "dst": 142
          },
          {
            "id": 145,
            "src": 142,
            "dst": 144
          },
          {
            "id": 147,
            "src": 144,
            "dst": 146
          },
          {
            "id": 149,
            "src": 146,
            "dst": 148
          },
          {
            "id": 151,
            "src": 148,
            "dst": 150
          }
        ]
      }
    }
  ],
  "contracts": []
}
