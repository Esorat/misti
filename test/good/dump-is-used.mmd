graph TD
subgraph test1
    test1_141["let a: Int = 42"]
    test1_142["dump(a)"]
    test1_144["dump(42)"]
    test1_146["dump('test is called')"]
    test1_148["dump(false)"]
    test1_150["dump(null)"]:::exitNode
    test1_141 --> test1_142
    test1_142 --> test1_144
    test1_144 --> test1_146
    test1_146 --> test1_148
    test1_148 --> test1_150
end
