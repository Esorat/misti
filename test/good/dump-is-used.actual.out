[INFO] DumpIsUsed: Found `dump` usage
test/good/dump-is-used.tact:4:5:
  3 |     let a: Int = 42;
> 4 |     dump(a);
          ^
  5 | 
Help: Using `dump` in production code can sometimes indicate complex code that requires additional review
See: https://nowarp.io/tools/misti/docs/detectors/DumpIsUsed